src/saturn_engine/utils/error_handling.py

```diff
@@ -0,0 +1,148 @@
+import os
+import re
+import sys
+from dataclasses import dataclass
+from functools import lru_cache
+from textwrap import dedent
+
+from saturn_engine.worker.executors.bootstrap import RemoteException
+
+
+@dataclass
+class ExceptionDetails:
+    message: str
+    exception_type: Exception
+    module: str
+    lineno: int
+
+
+@dataclass
+class ExceptionFilter(ExceptionDetails):
+    action: str
+
+
+def resolve_exception(exception_type: str) -> Exception:
+    if "." not in exception_type:
+        import builtins as m
+
+        klass = exception_type
+    else:
+        module, _, klass = exception_type.rpartition(".")
+        m = __import__(module, None, None, [klass])
+    return getattr(m, klass)
+
+
+def does_output_match(
+    exception: ExceptionDetails, exception_filter: ExceptionFilter
+) -> bool:
+    return (
+        (
+            exception_filter.message == ""
+            or re.match(exception_filter.message, exception.message) is not None
```

Comment:
Don't use `re.match`, `exception_filter.message` should be a `re.compile` regex and probably be renamed to `exception_filter.message_pattern` instead.